### PR TITLE
feat: trace_width option added

### PR DESCRIPTION
**Pull request by imrishabh18**
Didn't add the screenshot because the `soup-to-svg` is not parsing the `source_net` yet.

Closes - https://github.com/tscircuit/tscircuit/issues/264

### CODE REVIEW

File: src/lib/builder/net-builder/net-builder.ts [hunk 1]
```diff
@@ -51,6 +52,7 @@ export class NetBuilderClass implements NetBuilder {
         is_digital_signal: this.props.is_digital_signal,
         is_power: this.props.is_power,
         is_ground: this.props.is_ground,
+        trace_width: this.props.trace_width,
```
Comment (seveibar) [inline src/lib/builder/net-builder/net-builder.ts:55]: to fix this error, i think you'll need to add `trace_width` to `source_net` on [tscircuit/soup](https://github.com/tscircuit/soup)

Comment (seveibar) [reply src/lib/builder/net-builder/net-builder.ts:55]: ahh you did https://github.com/tscircuit/soup/pull/17#pullrequestreview-2210780920

**Review by seveibar — APPROVED** [review]
don't forget to commit the snapshot, would love to see it!

File: src/lib/builder/net-builder/net-builder.ts (hunk 1 above)
Comment (imrishabh18) [reply src/lib/builder/net-builder/net-builder.ts:55]: Yeah, have made another PR for it

**Comment by imrishabh18** [comment]
@seveibar Why does the pcb don't render anything for the net-builder-1 test https://debug.tscircuit.com/soup_group/builder:%20net%20builder%201#%7B%22selected_engine%22%3A%22pcb%22%2C%22selected_layout_index%22%3A0%7D

**Comment by seveibar** [comment]
@imrishabh18 yep thats fine

**Comment by imrishabh18** [comment]
The snapshot generated is also blank, like you can see in the debug view. So didn't add it to the pr

**Comment by seveibar** [comment]
@imrishabh18 might be a good idea to create a new test that has output.

Btw I have no idea why you're getting those type errors- I would make sure that @tscircuit/soup is a `peerDependency` and `dependency` in `circuit-to-svg` and `circuit-to-png`. I think if it's listed as a `devDependecy` OR isn't listed as a dependency then it can be accidentally bundled.

CC @andrii-balitskyi and I discussed linting that ensured that tscircuit dependencies don't get accidentally bundled

**Comment by imrishabh18** [comment]
Yeah, fixed it now. Added it as a peerDependency

**Comment by seveibar** [comment]
Oh @imrishabh18 sorry I read your comments to quick, I'm not sure why it's not outputing but it definitely seems like an issue we should investigate.

This is also making me think we need to add automatic scaling and `transformation-matrix` to `circuit-to-svg` because in this case the circuit is incredibly small (less than 5mm) so it needs to autoscale to fit the view like pcb-viewer. We should use `transformation-matrix` which makes this scaling really easy, you just do `transform.applyToPoint` and it will make each point in the correct location

**Comment by imrishabh18** [comment]
Okay, will take a look into `transformation-matrix`

**Pull request merged**